File: Week 07/id_533/LeetCode_338.js
// https://leetcode-cn.com/problems/counting-bits/

/**
 * @param {number} num
 * @return {number[]}
 */
var countBits = function(num) {
    var countArr = []
    for (var i = 0; i <= num; i++) {
        countArr.push(count(i))
    }
    return countArr;
};

function count(num) {
    var count = 0;
    while (num) {
        count++;
        num = num & (num - 1);
    }
    return count;
}

console.log(countBits(2))
console.log(countBits(5))

File: Week 07/id_113/LeetCode338.py
class Solution(object):
    def countBits(self, num):
        """
        :type num: int
        :rtype: List[int]
        """
        ans = []
        for n in range(num + 1):
            count = 0
            while n:
                count += n & 1
                n >>= 1
            ans.append(count)

        return ans